test/chrome/chromeDebugAdapter.test.ts

```diff
@@ -427,6 +427,30 @@ suite('ChromeDebugAdapter', () => {
             });
         });
 
+        // This is needed for Edge debug adapter, please keep the signature and logic of sendLoadedSourceEvent() method intact.
+        test('tests that sendLoadedSourceEvent can accept an additional paramter to override the `reason` parameter', done => {
+            const loadedSourceEventReason: LoadedSourceEventReason = 'changed';
+            sendEventHandler = (event) => {
+                assert.equal('loadedSource', event.event);
+                assert.notEqual(null, event.body);
+                assert.equal(loadedSourceEventReason, event.body.reason);
+                done();
```

Comment:
How will done be called if the test fails?

Reply:
It will timeout and show as a timeout failure.

Reply:
It'd be great if we could show the actual failure rather than a timeout.

Reply:
Updated to return the promise directly instead of using `done` method.